src/connectors/voice-search/__tests__/connectVoiceSearch-test.js

```diff
@@ -0,0 +1,188 @@
+import jsHelper, { SearchParameters } from 'algoliasearch-helper';
+import connectVoiceSearch from '../connectVoiceSearch';
+
+jest.mock('../../../lib/voiceSearchHelper', () => {
+  return ({ onStateChange, onQueryChange }) => {
+    return {
+      getState: () => {},
+      isBrowserSupported: () => true,
+      isListening: () => false,
+      toggleListening: () => {},
+      // ⬇️ for test
+      changeState: () => onStateChange(),
+      changeQuery: query => onQueryChange(query),
+    };
+  };
+});
+
+function getDefaultSetup() {
+  const renderFn = jest.fn();
+  const unmountFn = jest.fn();
+  const makeWidget = connectVoiceSearch(renderFn, unmountFn);
+  const widget = makeWidget({});
+  const helper = jsHelper({});
+
+  return {
+    renderFn,
+    unmountFn,
+    widget,
+    helper,
+  };
+}
+
+function getInitializedWidget() {
+  const { renderFn, unmountFn, widget, helper } = getDefaultSetup();
+
+  helper.search = () => {};
+  widget.init({ helper });
+
+  return {
+    renderFn,
+    unmountFn,
+    widget,
+    helper,
+    refine: query => widget._refine(query),
+  };
+}
+
+describe('connectVoiceSearch', () => {
+  describe('Usage', () => {
+    it('throws without render function', () => {
+      expect(() => {
+        connectVoiceSearch()({});
+      }).toThrowErrorMatchingInlineSnapshot(`
+"The render function is not valid (got type \\"undefined\\").
+
+See documentation: https://www.algolia.com/doc/api-reference/widgets/voice-search/js/#connector"
+`);
+    });
+  });
+
+  describe('lifecycle', () => {
+    it('calls renderFn during init and render', () => {
+      const { renderFn, widget, helper } = getDefaultSetup();
+      widget.init({ helper });
+      expect(renderFn).toHaveBeenCalled();
```

Comment:
Let's be more strict.

```suggestion
      expect(renderFn).toHaveBeenCalledTimes(1);
```